docs/guide/custom-functions.md

```diff
@@ -46,25 +71,79 @@ import { FunctionPlugin } from 'hyperformula';
 // start creating a class
 class CountHF extends FunctionPlugin {
 
-// define functions inside this plugin
+  // define functions inside this plugin
   public static implementedFunctions = {
     'HYPER': {
-    // this method's functionality will be defined in the next step
+      // this method's functionality will be defined below
       method: 'hyper',
-    // optionally, mark your function as volatile
+      // set your optional parameters below
+      arrayFunction: false,
+      doesNotNeedArgumentsToBeComputed: false,
+      expandRanges: false,
+      isDependentOnSheetStructureChange: false,
       isVolatile: true,
+      repeatLastArgs: 4,
+      vectorizationForbidden: false,
+    }
+  };
+}
+```
+
+You can set the following optional parameters:
+
+| Option | Type | Description |
+| --- | --- | --- |
+| `arrayFunction` | Boolean | If set to `true`, the function enables the [array arithmetic mode](arrays.md) in its arguments and nested expressions. |
+| `doesNotNeedArgumentsToBeComputed` | Boolean | If set to `true`, the function treats reference or range arguments as arguments that don't create dependency.<br><br>Other arguments are properly evaluated. |
+| `expandRanges` | Boolean | If set to `true`, ranges in the function's arguments are inlined to (possibly multiple) scalar arguments. |
+| `isDependentOnSheetStructureChange` | Boolean | If set to `true`, the function gets recalculated with each sheet shape change. |
+| `isVolatile` | Boolean | If set to `true`, the function is [volatile](volatile-functions.md). |
+| `repeatLastArgs` | Number | For functions with a variable number of arguments: sets how many last arguments can be repeated indefinitely. |
+| `vectorizationForbidden` | Boolean | If set to `true`, prevents the function from ever being vectorized (but it's up to your function implementation to properly handle vectorization). |
```

Comment:
We do not explain what does the vectorization means and what will happen if it's not handled by user 🤔 or how to do it.

Reply:
@wojciechczerniak - that's right (this bit is taken from an old description of this option).

I [added a link](https://github.com/handsontable/hyperformula/pull/758/commits/195c9a15db173963ecb6222dcd2193a07860349b#diff-aeaf0b9b496edad5d31a7315f0be860167daf08de1368a16e6d4b7f2bf81badeR102) to the only place where vectorization is mentioned so far, a [section](https://github.com/handsontable/hyperformula/blob/feature/issue-740/docs/guide/arrays.md#passing-arrays-to-scalar-functions) in the new "Arrays" topic.

I'm guessing that to properly describe vectorization, we'd need a separate section in the Arrays topic, or even a completely separate topic. Do we want to do that now? I guess it depends on what possible implications are - if it's vital to "properly handle vectorization" when implemenitng any sort of custom function, we should add it now, if not, we could do it later on

Reply:
No, we don't have to. It's a lot better than outdated custom functions page and will be much more helpful even without all the possible details.

Reply:
It is tricky -- `vectorizationForbidden` is an internal flag, that is used by our internal wrapper function. Any external function is free to ignore it.

Reply:
(in fact, right now external functions cannot return arrays correctly)